components/SetAction.coffee

```diff
@@ -0,0 +1,25 @@
+noflo = require 'noflo'
+
+exports.getComponent = ->
+  c = new noflo.Component
+  c.inPorts.add 'action',
+    datatype: 'string'
+    required: true
+    control: true
+  c.inPorts.add 'in',
+    datatype: 'all'
+  c.outPorts.add 'out',
+    datatype: 'all'
+
+  noflo.helpers.WirePattern c,
+    params: ['action']
+    forwardGroups: false
+    async: true
+  , (data, groups, out, callback) ->
+    actionParts = c.params.action.split ':'
+    setTimeout ->
```

Comment:
Workaround for bracket forwarding bug in WirePattern. Would be fixed by going 0.8 process API